spring-kafka-test/src/main/java/org/springframework/kafka/test/EmbeddedKafkaKraftBroker.java

```diff
@@ -225,14 +219,17 @@ private void start() {
 		}
 
 		createKafkaTopics(this.topics);
-		if (this.brokerListProperty == null) {
-			this.brokerListProperty = System.getProperty(BROKER_LIST_PROPERTY);
-		}
+
+		String brokersAsString = getBrokersAsString();
 		if (this.brokerListProperty != null) {
-			System.setProperty(this.brokerListProperty, getBrokersAsString());
+			System.setProperty(this.brokerListProperty, brokersAsString);
+		}
+		String globalBrokerListProperty = System.getProperties().containsKey(BROKER_LIST_PROPERTY)
+				? System.getProperties().getProperty(BROKER_LIST_PROPERTY)
+				: SPRING_EMBEDDED_KAFKA_BROKERS;
+		if (!globalBrokerListProperty.equals(this.brokerListProperty)) {
+			System.setProperty(globalBrokerListProperty, brokersAsString);
```

Comment:
we have two ways to pass the broker list property:

- chaining setter
- system property

Seems both are working independently; for that reason, I renamed the latter as 'globalBrokerListProperty` to differentiate with `this.brokerListProperty`.